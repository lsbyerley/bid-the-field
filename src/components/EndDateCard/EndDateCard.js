import { CalendarIcon } from '@heroicons/react/24/outline';
import { format } from 'date-fns';
import useAsyncReference from '@/lib/useAsyncReference';

const EndDateCard = ({ auction, auctionOver }) => {
  const asyncAuction = useAsyncReference(auction, true);

  return (
    <div className='rounded-lg card compact bg-base-100'>
      <div className='relative justify-center card-body'>
        {auctionOver && (
          <div className='absolute badge badge-error badge-outline top-5 right-5'>
            Bidding Over
          </div>
        )}
        {!auctionOver && (
          <div className='absolute badge badge-success badge-outline top-5 right-5'>
            In Progress
          </div>
        )}
        <div className='flex items-center'>
          <div className='flex-shrink-0'>
            <CalendarIcon
              className='w-6 h-6 text-gray-400'
              aria-hidden='true'
            />
          </div>
          <div className='flex-1 w-0 ml-5'>
            <dl>
              <dt className='text-sm font-medium truncate'>End Date</dt>
              <dd>
                <div className='font-medium '>
                  {format(
                    new Date(asyncAuction.current.end_date),
                    'LLL d, h:mm aaa'
                  )}
                </div>
              </dd>
            </dl>
          </div>
        </div>
      </div>
    </div>
  );
};

export default EndDateCard;
