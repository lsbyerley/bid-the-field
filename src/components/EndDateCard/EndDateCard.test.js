import React from 'react';
import { render, screen } from 'test-utils';
import EndDateCard from './EndDateCard';

jest.mock('date-fns', () => ({
  format: jest.fn((props) => JSON.stringify(props)),
}));

const defaultProps = (overrides) => {
  return {
    auction: {
      end_date: 'test',
    },
    auctionOver: false,
    ...overrides,
  };
};

const renderEndDateCard = (props) => {
  const utils = render(<EndDateCard {...props} />);
  const home = utils.container.firstChild;
  return { ...utils, home };
};

describe('EndDateCard', () => {
  it('should render and match snapshot', () => {
    const { home } = renderEndDateCard(defaultProps());
    expect(home).toBeTruthy();
    expect(home).toMatchSnapshot();
  });
});
