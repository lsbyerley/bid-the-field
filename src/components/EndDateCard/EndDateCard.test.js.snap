// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndDateCard should render and match snapshot 1`] = `
<div
  class="rounded-lg card compact bg-base-100"
>
  <div
    class="relative justify-center card-body"
  >
    <div
      class="absolute badge badge-success badge-outline top-5 right-5"
    >
      In Progress
    </div>
    <div
      class="flex items-center"
    >
      <div
        class="flex-shrink-0"
      >
        <svg
          aria-hidden="true"
          class="w-6 h-6 text-gray-400"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="flex-1 w-0 ml-5"
      >
        <dl>
          <dt
            class="text-sm font-medium truncate"
          >
            End Date
          </dt>
          <dd>
            <div
              class="font-medium "
            >
              null
            </div>
          </dd>
        </dl>
      </div>
    </div>
  </div>
</div>
`;
